### PR TITLE
Make python3 compatible

### PR DESCRIPTION
**Pull request by oysteoh**
depends on https://github.com/equinor/komodo-releases/pull/13

we need to merge new release / repository files in first - otherwise our builds will break

### CODE REVIEW

File: bin/kmd [hunk 1]
```diff
@@ -27,12 +27,12 @@ def main(args):
     tmp_prefix = os.path.join(os.path.join(args.prefix), args.release, 'root')
     fakeroot = os.path.abspath(args.release)
     if args.build or not args.install:
-        komodo.make(args.pkgs, args.repo, prefix   = tmp_prefix,
-                                         dlprefix = args.cache,
-                                         builddir = args.tmp,
-                                         jobs     = args.jobs,
-                                         cmk      = args.cmake,
-                                         fakeroot = fakeroot)
+        komodo.make(args.pkgs, args.repo,   prefix   = tmp_prefix,
```
Comment (markusdregi) [inline bin/kmd]: Aligning things at non-standard levels of indenting always break down over time. How about:

```python
komodo.make(
    args.pkgs,
    args.repo,
    prefix=tmp_prefix,
    dlprefix=args.cache,
    builddir=args.tmp,
    jobs=args.jobs,
    cmk=args.cmake,
    fakeroot = fakeroot,
    )
```
```

File: bin/kmd [hunk 2]
```diff
@@ -97,17 +96,28 @@ def main(args):
                                    os.environ.get('PATH', '')])
 
     print('Fixup #! in pip-provided packages')
+    if 'python' in pkgs.keys():
+        # Currently we need to keep pip to 18.1 due to a bug in pip 19<
```
Comment (markusdregi) [inline bin/kmd]: Maybe this instead?
> due to a bug in pip >=19

File: bin/kmd [hunk 3]
```diff
@@ -97,17 +96,28 @@ def main(args):
...
+        shell_input = [ '{}/bin/pip'.format(install_root),
+                        'install {}=={}'.format(pkg, stripped_ver),
+                        '--prefix', os.path.join(args.prefix, args.release, 'root'),
+                        '--force-reinstall',
+                        '--no-index',
+                        '--no-deps',
+                        '--find-links {}'.format(args.cache),
+                        repo[pkg][ver].get('makeopts')]
+
+        print(komodo.shell(shell_input, sudo = args.sudo))
```
Comment (markusdregi) [inline bin/kmd]: `sudo=args.sudo`

File: komodo/build.py [hunk 4]
```diff
@@ -102,10 +102,12 @@ def rsync(pkg, ver, pkgpath, prefix, *args, **kwargs):
 
 def pip(pkg, ver, pkgpath, prefix, dlprefix, *args, **kwargs):
     # pip should be in the tmp root, which is already added to PATH
-    cmd = ['pip install {}=={}'.format(pkg, ver),
+    stripped_ver = ver.split("+")[0]
```
Comment (markusdregi) [inline komodo/build.py]: Should we isolate this into a function in the `komodo`-module?

File: bin/kmd (hunk 1 above)
Comment (markusdregi) [reply bin/kmd]: I see that my comments regarding white spacing and named arguments are not in line with the rest of the code and that the rest of the code is not in line with PEP-8. Do as you like ;)

File: bin/kmd (hunk 3 above)
Comment (markusdregi) [reply bin/kmd]: I see that my comments regarding white spacing and named arguments are not in line with the rest of the code and that the rest of the code is not in line with PEP-8. Do as you like ;)

**Comment by markusdregi** [comment]
Excellent work :+1:

File: bin/kmd (hunk 2 above)
Comment (oysteoh) [reply bin/kmd]: yes!

File: komodo/build.py [hunk 5]
```diff
@@ -198,7 +200,7 @@ def resolve(x):
 
     fakeroot = args.prefix if args.fakeroot else '.'
     args = parser.parse_args()
-    make(args.pkgfile, args.repofile, prefix = args.prefix,
+    make(args.pkgfile, args.repofile, prefix   = args.prefix,
```
Comment (markusdregi) [inline komodo/build.py]: Should we revert this one also?

**Comment by markusdregi** [comment]
Besides the white-space comment I'm ready to merge this now :rocket:

File: komodo/build.py (hunk 5 above)
Comment (oysteoh) [reply komodo/build.py]: yes, i just didn't notice that one ... now it's gone :+1:

**Pull request merged**